src/Eshopworld.Core/IMessenger.cs

```diff
@@ -25,17 +25,11 @@ public interface IPublishEvents : IDisposable
         /// <summary>
         /// Sends an event onto a topic.
         /// </summary>
-        /// <typeparam name="T">The type of the event that we are sending.</typeparam>
-        /// <param name="event">The event that we are sending.</param>
-        Task Publish<T>(T @event) where T : class;
-
-        /// <summary>
-        /// Sends an event onto a topic.
-        /// </summary>
+        /// <remarks>If <paramref name="topicName"/> is not provided then the event type is used to determine the topic</remarks>
         /// <typeparam name="T">The type of the event that we are sending.</typeparam>
         /// <param name="event">The event that we are sending.</param>
         /// <param name="topicName">The name of the topic we are sending to</param>
-        Task Publish<T>(T @event, string topicName) where T : class;
+        Task Publish<T>(T @event, string? topicName = null) where T : class;
```

Comment:
Do we need to mark is nullable here? I think it is not need for string type

Reply:
In c# 8.0 it gives you an error when you assign null to a reference type:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-reference-types

This is a change, and we should start marking as nullable reference types to express the intent that it can be set to null

Reply:
it gives you a warning sorry. not an error